crates/red_knot/src/target_version.rs

```diff
@@ -0,0 +1,50 @@
+use std::fmt;
+
+/// Enumeration of all supported Python versions
+///
+/// TODO: unify with the `PythonVersion` enum in the linter/formatter crates?
+#[derive(Copy, Clone, Hash, Debug, PartialEq, Eq, PartialOrd, Ord, Default, clap::ValueEnum)]
+pub enum TargetVersion {
```

Comment:
Why do we need this defined both here and in `red_knot_module_resolver` crate?

Reply:
I originally just imported it from `red_knot_module_resolver`, but Micha suggested duplicating it here so that we can avoid having `clap` as a dependency in `red_knot_module_resolver`: https://github.com/astral-sh/ruff/pull/12246#discussion_r1668970240. For the version of the enum that's used as a `clap` argument, it needs to have `ValueEnum` derived on it, which means that the crate in which it's defined needs to have `clap` as a dependency